ddl/partition.go

```diff
@@ -1102,18 +1102,18 @@ func onTruncateTablePartition(d *ddlCtx, t *meta.Meta, job *model.Job) (int64, e
 	if d.infoHandle != nil && d.infoHandle.IsValid() {
 		bundles := make([]*placement.Bundle, 0, len(oldIDs))
 
-		yoldIDs := make([]int64, 0, len(oldIDs))
 		newIDs := make([]int64, 0, len(oldIDs))
 		for i, oldID := range oldIDs {
 			oldBundle, ok := d.infoHandle.Get().BundleByName(placement.GroupID(oldID))
 			if ok && !oldBundle.IsEmpty() {
-				yoldIDs = append(yoldIDs, oldID)
 				newIDs = append(newIDs, newPartitions[i].ID)
 				bundles = append(bundles, placement.BuildPlacementDropBundle(oldID))
 				bundles = append(bundles, placement.BuildPlacementCopyBundle(oldBundle, newPartitions[i].ID))
+			} else {
+				newIDs = append(newIDs, -1)
 			}
 		}
-		job.CtxVars = []interface{}{yoldIDs, newIDs}
+		job.CtxVars = []interface{}{oldIDs, newIDs}
```

Comment:
How about this:
oldIDs = old partition IDs
newIDs = new partition IDs
In applyDiff, always drop placement rules for old partitions, and always fetch placement rules for new partitions, no matter it exists or not.

Reply:
So if you truncate 10 partitions, you got 10 rtt of  TiDB-PD.